src/components/Confirm/index.js

```diff
@@ -98,15 +101,148 @@ const Previews = ({capture, retakeAction, confirmAction, error} ) =>
     <Actions retakeAction={retakeAction} confirmAction={confirmAction} error={error} />
   </div>
 
-const Confirm = ({method, side, validCaptures:[capture], onConfirm, error, onRetake, actions: {deleteCaptures}}) =>
-  <Previews
-    capture={capture}
-    retakeAction={() => {
-      deleteCaptures({method, side})
-      onRetake()
-    }}
-    confirmAction={onConfirm}
-    error={error}
-  />
-
-export default trackComponentAndMode(Confirm, 'confirmation', 'error')
+const ProcessingApiRequest = () =>
+  <div className={theme.center}>
+    <Spinner />
+  </div>
+
+class Confirm extends Component  {
+
+  constructor(props){
+    super(props)
+    this.state = {
+      uploadInProgress: false,
+      error: {},
+      captureId: null,
+      onfidoId: null,
+     }
+  }
+
+  clearErrors = () => {
+    this.setState({error: {}})
+  }
+
+  onGlareWarning = () => {
+    this.setWarning('GLARE_DETECTED')
+  }
+
+  setError = (name) => this.setState({error: {name, type: 'error'}})
+  setWarning = (name) => this.setState({error: {name, type: 'warn'}})
+
+  onfidoErrorFieldMap = ([key, val]) => {
+    if (key === 'document_detection') return 'INVALID_CAPTURE'
+    // on corrupted PDF or other unsupported file types
+    if (key === 'file') return 'INVALID_TYPE'
+    // hit on PDF/invalid file type submission for face detection
+    if (key === 'attachment' || key === 'attachment_content_type') return 'UNSUPPORTED_FILE'
+    if (key === 'face_detection') {
+      return val[0].indexOf('Multiple faces') === -1 ? 'NO_FACE_ERROR' : 'MULTIPLE_FACES_ERROR'
+    }
+  }
+
+  onfidoErrorReduce = ({fields}) => {
+    const [first] = Object.entries(fields).map(this.onfidoErrorFieldMap)
+    return first
+  }
+
+  onApiError = ({status, response}) => {
+    let errorKey;
+    if (status === 422){
+      errorKey = this.onfidoErrorReduce(response.error)
+    }
+    else {
+      sendError(`${status} - ${response}`)
+      errorKey = 'SERVER_ERROR'
+    }
+
+    this.setState({uploadInProgress: false})
+    this.setError(errorKey)
+  }
+
+  onApiSuccess = (apiResponse) => {
+    this.setState({onfidoId: apiResponse.id})
+    const warnings = apiResponse.sdk_warnings
+    if (warnings && !warnings.detect_glare.valid) {
+      this.onGlareWarning()
+    }
+    else {
+      this.confirmAndProceed(apiResponse)
+    }
+    this.setState({uploadInProgress: false})
+  }
+
+  confirmEvent = (method, side) => {
+    if (method === 'document') {
+      if (side === 'front') events.emit('documentCapture')
+      else if (side === 'back') events.emit('documentBackCapture')
+    }
+    else if (method === 'face') events.emit('faceCapture')
+  }
+
+  confirmAndProceed = () => {
+    const {method, side, nextStep, actions: {confirmCapture}} = this.props
+    this.clearErrors()
```

Comment:
I've moved this logic in the `onUploadFallback` function within the capture component, to clear up errors when transitioning from webcam to upload fallback.